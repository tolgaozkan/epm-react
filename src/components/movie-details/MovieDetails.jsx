import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

require('./style.scss');

export class MovieDetails extends React.PureComponent {
  render() {
    const { movies } = this.props;
    const movie = movies.find(m => m.id === parseInt(this.props.match.params.id, 10));

    return movie ? (
      <div className="movie-detail-container">
        <div className="movie">
          <p className="movie-name">Name: {movie.title}</p>
          <p className="movie-genre">Genres: {movie.genres.join(', ')}</p>
        </div>
      </div>
    ) : (<div />);
  }
}

MovieDetails.propTypes = {
  movies: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    movies: state.movies,
  };
}

export default connect(mapStateToProps)(MovieDetails);
