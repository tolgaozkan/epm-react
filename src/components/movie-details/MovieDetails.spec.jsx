import * as React from 'react';
import { create as render } from 'react-test-renderer';

import { MovieDetails } from './MovieDetails';

it('renders movie details', () => {
  const movies = [
    {
      id: 1,
      title: 'Guardians of the Galaxy Vol. ',
      genres: ['Action', 'Adventure', 'Science Fiction'],
      release_date: '2020-05-01',
      overview: "The third film based on Marvel's Guardians of the Galaxy.",
    },
    {
      id: 2,
      title: 'Transformers 7',
      genres: ['Science Fiction', 'Action', 'Adventure'],
      release_date: '2019-06-26',
    },
  ];

  const match = {
    params: {
      id: '1',
    },
  };

  const result = render(<MovieDetails movies={movies} match={match} />);
  const snap = result.toJSON();

  expect(snap).toMatchSnapshot();
});
