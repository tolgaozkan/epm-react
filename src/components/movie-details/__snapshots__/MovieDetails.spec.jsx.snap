// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movie details 1`] = `
<div
  className="movie-detail-container"
>
  <div
    className="movie"
  >
    <p
      className="movie-name"
    >
      Name: 
      Guardians of the Galaxy Vol. 
    </p>
    <p
      className="movie-genre"
    >
      Genres: 
      Action, Adventure, Science Fiction
    </p>
  </div>
</div>
`;
